Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/DriverFeeUpdates/DriverFee.hs

```diff
@@ -410,57 +405,194 @@ mkInvoiceAgainstDriverFee driverFee (isCoinCleared, isAutoPay) = do
+  let jobData = jobInfo.jobData
+      merchantId = jobData.merchantId
+      opCityId = jobData.merchantOperatingCityId
+      startTime = jobData.startTime
+      endTime = jobData.endTime
+      serviceName = jobData.serviceName
+  now <- getCurrentTime
+  subscriptionConfigs <- CQSC.findSubscriptionConfigsByMerchantOpCityIdAndServiceName opCityId serviceName >>= fromMaybeM (InternalError $ "No subscription config found" <> show serviceName)
+  let deepLinkExpiry = subscriptionConfigs.deepLinkExpiryTimeInMinutes
+  driverFeesToProccess <- QDF.findDriverFeeInRangeEligibleForManualPaymentLinkByServiceName merchantId subscriptionConfigs.genericBatchSizeForJobs startTime endTime jobData.serviceName
+  if null driverFeesToProccess
+    then return Complete
+    else do
+      processAndSendManualPaymentLink driverFeesToProccess subscriptionConfigs merchantId opCityId serviceName deepLinkExpiry now
+      ReSchedule <$> getRescheduledTime subscriptionConfigs.genericJobRescheduleTime
+
+processAndSendManualPaymentLink :: (EsqDBReplicaFlow m r, EsqDBFlow m r, EncFlow m r, CacheFlow m r, HasField "smsCfg" r SmsConfig) => [DriverFee] -> SubscriptionConfig -> Id Merchant -> Id MerchantOperatingCity -> ServiceNames -> Maybe Int -> UTCTime -> m ()
+processAndSendManualPaymentLink driverFeesToProccess subscriptionConfigs merchantId opCityId serviceName mbDeepLinkExpiry now = do
+  forM_ driverFeesToProccess $ \driverFeeForManualCharge -> do
+    QDF.updateStatus PAYMENT_OVERDUE (driverFeeForManualCharge.id) now
+    let driverId = driverFeeForManualCharge.driverId
+    let mbDeepLinkData = Just $ SPayment.DeepLinkData {sendDeepLink = Just True, expiryTimeInMinutes = mbDeepLinkExpiry}
+    resp' <- try @_ @SomeException $ DDriver.clearDriverDues (driverId, merchantId, opCityId) serviceName mbDeepLinkData
+    errorCatchAndHandle
+      driverFeeForManualCharge
+      resp'
+      subscriptionConfigs
+      now
+      ( \resp -> do
+          let mbPaymentLink = resp.orderResp.payment_links
+              payload = resp.orderResp.sdk_payload.payload
+              mbAmount = readMaybe (T.unpack payload.amount) :: Maybe HighPrecMoney
+          whatsAppResp <- try @_ @SomeException $ SPayment.sendLinkTroughChannelProvided mbPaymentLink driverId mbAmount (Just subscriptionConfigs.paymentLinkChannel) WHATSAPP_SEND_MANUAL_PAYMENT_LINK
+          errorCatchAndHandle driverFeeForManualCharge whatsAppResp subscriptionConfigs now (\_ -> return ())
+      )
+
+errorCatchAndHandle :: (EsqDBReplicaFlow m r, EsqDBFlow m r, EncFlow m r, CacheFlow m r, HasField "smsCfg" r SmsConfig) => DriverFee -> Either SomeException a -> SubscriptionConfig -> UTCTime -> (a -> m ()) -> m ()
+errorCatchAndHandle driverFeeForManualCharge resp' subscriptionConfig now function = do
+  case resp' of
+    Left _ -> do
+      eligibleForRetryInNextBatch <- redisBasedErrorConter (mkManualLinkErrorTrackingKey driverFeeForManualCharge.id) subscriptionConfig.maxRetryCount
```

Comment:
`redisBasedErrorCounter` -> `isEligibleForRetry`, or something like that would be better IMO.